portfolio/src/main/webapp/script.js

```diff
@@ -82,3 +82,9 @@ function getComments() {
       }
     });
 }
+
+function submitComment(){
+  const newComment = document.getElementById('new-comment').value;
+  const request = new Request(`/data?new-comment=${newComment}`, {method:'POST'});
```

Comment:
I believe Request has a nicer way to add parameters.

https://developer.mozilla.org/en-US/docs/Web/API/Request

Something along the lines of
`const request = new Request('https://example.com', {method: 'POST', body: '{"foo": "bar"}'});`

That would make it easier to read.

Reply:
The `response.getParameter("new-comment")` can be safely used when:

1. the parameter is included in a <form> element with `action='/data'` and `<input type="submit">` set up. HTML's `formhandler` will nicely [set up the field](https://metacpan.org/pod/HTML::FormHandler#Accessing-and-setting-up-fields).
2. the parameter is passed in the query string

However when parameters are sent in the request body, `getParameter("new-comment")` somehow always returns `null`. I think it was [affected by other IO methods](https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getParameter(java.lang.String)) such as `getReader()`. I tried to use `getReader()` instead but it involves quite [some conversions and parsing with external libraries](https://stackoverflow.com/questions/8100634/get-the-post-request-body-from-httpservletrequest/18632784) to extract the parameter from the query. 

Since there is only one parameter here, sticking with the query string makes most sense to me. If you can find some easy way to extract the parameters from the request body, I am excited to learn!